src/main/java/seedu/address/model/person/Person.java

```diff
@@ -149,14 +149,15 @@ public Person withMeetings(List<Event> meetings) {
      * Compute the next goal deadline using meetings before the {@code beforeDate}
      */
     public LocalDate getGoalDeadline(LocalDate beforeDate) {
+        System.out.println(meetings);
         if (meetings.isEmpty()) {
             // user has never met up with this person before
             return DateUtil.ZERO_DAY;
         }
 
         LocalDate latestMeetingDate = meetings.stream()
                 .map(Event::getDate)
-                .filter(x -> x.isBefore(beforeDate))
+                .filter(x -> x.isBefore(beforeDate.plusDays(1)))
```

Comment:
This is to account for the meeting added today? Write comment above saying that the `latestMeetingDate` includes `today`